Neos.Flow/Classes/Http/Request.php

```diff
@@ -140,7 +147,72 @@ public function __construct(array $get, array $post, array $files, array $server
         $this->server = $server;
         $untangledFiles = UploadedFilesHelper::untangleFilesArray($files);
         $this->arguments = ArgumentsHelper::buildUnifiedArguments($get, $post, $untangledFiles);
-        $this->uploadedFiles = $this->createUploadedFilesFromUntangledUploads($untangledFiles, $this->arguments);
+        $this->uploadedFiles = static::createUploadedFilesFromUntangledUploads($untangledFiles, $this->arguments);
+    }
+
+    /**
+     * Constructs a new Request object based on the given environment data.
+     *
+     * @param array $get Data similar to that which is typically provided by $_GET
+     * @param array $post Data similar to that which is typically provided by $_POST
+     * @param array $files Data similar to that which is typically provided by $_FILES
+     * @param array $server Data similar to that which is typically provided by $_SERVER
+     * @return ServerRequestInterface
+     * @see create()
+     * @see createFromEnvironment()
+     */
+    private static function _create(array $get, array $post, array $files, array $server): ServerRequestInterface
```

Comment:
For now solved the main issue (`Request::create[FromEnvironment]()` all around the places) by just creating a factory method that returns an `ServerRequestInterface` instance. The other solution would be, to make our `Http\Request` decorate on an actual exchangeable `ServerRequestInterface` (and provide some convenience helpers on top). I mean, the main reason for us to use PSR-7 (and PSR-15) is to be able to make use of existing middlewares and keep the HTTP Kernel implementation outside of Flow, right? But that is still guaranteed, even if inside Flow we work with an Http Request class, that provides a bit more than plain PSR-7 (e.g. `getBaseUri()`, `getArguments()`, Headers and Cookies objects that provide some useful IDE help). WDYT?